lms/services/course.py

```diff
@@ -205,10 +223,7 @@ def find_group_set(self, group_set_id=None, name=None, context_id=None):
         return None
 
     def get_by_id(self, id_: int) -> Course | None:
-        if courses := self.search(id_=id_, limit=1):
-            return courses[0]
-
-        return None
+        return self._search_query(id_=id_).one_or_none()
```

Comment:
Having the query method allows us to simplify this.

Reply:
Does `one_or_none()` implicitly add a LIMIT 1 to the query?

Reply:
Oh, missed this before merging.

No it doesn't it will execute the query as it is and raise if there's more than one result. 

This is a query on the PK so limit should make any perf difference.